### PR TITLE
Fix problem creating codespaces environment

### LINKED ISSUE #14750: Codespaces is broken
@taldcroft says Codespaces that Microsoft help us set up is now broken.

### PR DESCRIPTION
**Pull request by taldcroft**
### Description

This pull request is to try fixing a problem creating codespaces. What was happening is that for some reason the full list of repository tags was not being pulled down, despite the `git fetch --tag`. Then when running the `pip install -e .[doc,test-all,docs]` it thought it was installing `astropy-0.1-xxx` and some packages that depend on a minimum astropy version failed.

I was able to reproduce this in my fork, although it was getting astropy 4.1-xxx. I found that using `git fetch --all` fixed the problem in my fork, so giving it a try here as a PR.

Fixes #14750

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "name": "AstroPy",
     "image": "mcr.microsoft.com/devcontainers/miniconda:0-3",
     "onCreateCommand": "conda init bash && sudo cp .devcontainer/welcome-message.txt /usr/local/etc/vscode-dev-containers/first-run-notice.txt",
-    "postCreateCommand": "git fetch --tag && pip install tox && pip install -e .[all,test_all,docs]",
+    "postCreateCommand": "git fetch --all && pip install tox && pip install -e .[all,test_all,docs]",
```
Comment (saimn) [inline .devcontainer/devcontainer.json:5]: The correct option (from git's manpage) is `--tags`, not `--tag`.

Comment (taldcroft) [reply .devcontainer/devcontainer.json:5]: Makes sense, except creating codespaces *did* work at some point in the past. And now using `--all` is failing when running on the astropy repo while it worked on my local fork. Ugh.

**Comment by saimn** [comment]
It's still installing astropy-0.1.dev, but maybe because it's cloning from your fork which does not have all the tags. And it fails because "asdf-astropy 0.4.0 depends on astropy>=5.0.4".
We can probably merge and see if it works on the main repo.

**Comment by taldcroft** [comment]
About the local fork tags, I was on the same page and trying to fix that. I synced fork on GitHub, which didn't quite help, then fetched all the tags locally and tried to push them up:
```
(astropy) ➜  astropy git:(main) git push origin main --tags
Enumerating objects: 22350, done.
Counting objects: 100% (17879/17879), done.
Delta compression using up to 16 threads
Compressing objects: 100% (4903/4903), done.
Writing objects: 100% (12647/12647), 3.32 MiB | 2.16 MiB/s, done.
Total 12647 (delta 9799), reused 10301 (delta 7687), pack-reused 0
```
It's just been sitting there for a few minutes now. Huh.

**Comment by taldcroft** [comment]
Anyway @saimn given the scope here if you want to Approve then I think there is a decent chance that the fix will work once this is merged to `main`.

**Review by saimn — APPROVED** [review]
Yes, let's merge and see if it works on main :+1:

**Comment by pllim** [comment]
So did it work?

**Pull request merged**